pkg/interface/src/components/Omnibox.js

```diff
@@ -0,0 +1,241 @@
+import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
+import { Box, Row, Rule, Text } from '@tlon/indigo-react';
+import index from '../lib/omnibox';
+import Mousetrap from 'mousetrap';
+import OmniboxResult from './OmniboxResult';
+
+import { cite } from '../lib/util';
+
+export class Omnibox extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      index: new Map([]),
+      query: '',
+      results: this.initialResults(),
+      selected: ''
+    };
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+    this.search = this.search.bind(this);
+    this.navigate = this.navigate.bind(this);
+    this.control - this.control.bind(this);
+    this.setPreviousSelected = this.setPreviousSelected.bind(this);
+    this.setNextSelected = this.setNextSelected.bind(this);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps !== this.props) {
+      this.setState({ index: index(this.props.associations, this.props.apps.tiles) });
+    }
+
+    if (prevProps && this.props.show && prevProps.show !== this.props.show) {
+      Mousetrap.bind('escape', () => this.props.api.local.setOmnibox());
+      document.addEventListener('mousedown', this.handleClickOutside);
+      this.input.focus();
```

Comment:
focus on show doesn't work on safari

Reply:
Fun fact: Safari needs a `touchstart` event first. Well. OK. Apple just wants me to add another line here.